src/bpf/hidrecord.skel.rs

```diff
@@ -0,0 +1,693 @@
+        }
+
+        pub fn hid_record(&self) -> &libbpf_rs::Map {
+            self.inner.map("hid_record").unwrap()
+        }
+    }
+
+    pub struct HidrecordProgs<'a> {
+        inner: &'a libbpf_rs::Object,
+    }
+
+    impl HidrecordProgs<'_> {
+        pub fn hid_record_event(&self) -> &libbpf_rs::Program {
+            self.inner.prog("hid_record_event").unwrap()
+        }
+    }
+
+    pub struct HidrecordProgsMut<'a> {
+        inner: &'a mut libbpf_rs::Object,
+    }
+
+    impl HidrecordProgsMut<'_> {
+        pub fn hid_record_event(&mut self) -> &mut libbpf_rs::Program {
+            self.inner.prog_mut("hid_record_event").unwrap()
+        }
+    }
+
+    #[derive(Default)]
+    pub struct HidrecordLinks {
+        pub hid_record_event: Option<libbpf_rs::Link>,
+    }
+
+    pub struct HidrecordSkel<'a> {
+        pub obj: libbpf_rs::Object,
+        struct_ops: hidrecord_types::struct_ops,
+        skel_config: libbpf_rs::__internal_skel::ObjectSkeletonConfig<'a>,
+        pub links: HidrecordLinks,
+    }
+
+    unsafe impl Send for HidrecordSkel<'_> {}
+    unsafe impl Sync for HidrecordSkel<'_> {}
+
+    impl Skel for HidrecordSkel<'_> {
+        fn object(&self) -> &libbpf_rs::Object {
+            &self.obj
+        }
+
+        fn object_mut(&mut self) -> &mut libbpf_rs::Object {
+            &mut self.obj
+        }
+
+        fn attach(&mut self) -> libbpf_rs::Result<()> {
+            let ret = unsafe { libbpf_sys::bpf_object__attach_skeleton(self.skel_config.get()) };
+            if ret != 0 {
+                return Err(libbpf_rs::Error::from_raw_os_error(-ret));
+            }
+
+            self.links = HidrecordLinks {
+                hid_record_event: core::ptr::NonNull::new(self.skel_config.prog_link_ptr(0)?)
+                    .map(|ptr| unsafe { libbpf_rs::Link::from_ptr(ptr) }),
+            };
+
+            Ok(())
+        }
+    }
+    impl HidrecordSkel<'_> {
+        pub fn progs_mut(&mut self) -> HidrecordProgsMut<'_> {
+            HidrecordProgsMut {
+                inner: &mut self.obj,
+            }
+        }
+
+        pub fn progs(&self) -> HidrecordProgs<'_> {
+            HidrecordProgs { inner: &self.obj }
+        }
+
+        pub fn maps_mut(&mut self) -> HidrecordMapsMut<'_> {
+            HidrecordMapsMut {
+                inner: &mut self.obj,
+            }
+        }
+
+        pub fn maps(&self) -> HidrecordMaps<'_> {
+            HidrecordMaps { inner: &self.obj }
+        }
+
+        pub fn struct_ops_raw(&self) -> *const hidrecord_types::struct_ops {
+            &self.struct_ops
+        }
+
+        pub fn struct_ops(&self) -> &hidrecord_types::struct_ops {
+            &self.struct_ops
+        }
+    }
+
+    const DATA: &[u8] = &[
+        127, 69, 76, 70, 2, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 247, 0, 1, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 40, 28, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 64, 0, 0, 0, 0,
```

Comment:
I'm getting a lot of diff in these files as soon as I run cargo build. It's just bytes but also means I need to be careful with git commit, git diff is messy, etc.

Haven't tracked down why yet but it's probably a version mismatch but either way it's annoying. git doesn't seem to have an option for "track this file but don't change it".

Reply:
done in #34, please feel free to cherry-pick and squash